src/client/input/sdl.c

```diff
@@ -1360,6 +1382,379 @@ In_FlushQueue(void)
+{
+	int i, haptic_volume;
+
+	// check effects for reuse
+	for (i=0; i < last_haptic_effect_size; i++)
+	{
+		if (
+			last_haptic_effect[i].effect_volume != effect_volume ||
+			last_haptic_effect[i].effect_duration != effect_duration ||
+			last_haptic_effect[i].effect_delay != effect_delay ||
+			last_haptic_effect[i].effect_attack != effect_attack ||
+			last_haptic_effect[i].effect_fade != effect_fade ||
+			last_haptic_effect[i].effect_x != effect_x ||
+			last_haptic_effect[i].effect_y != effect_y ||
+			last_haptic_effect[i].effect_z != effect_z)
+		{
+			continue;
+		}
+
+		return last_haptic_effect[i].effect_id;
+	}
+
+	/* create new effect */
+	haptic_volume = joy_haptic_magnitude->value * effect_volume; // 32767 max strength;
+
+	/*
+	Com_Printf("%d: volume %d: %d ms %d:%d:%d ms speed: %.2f\n",
+		last_haptic_effect_pos,  effect_volume, effect_duration,
+		effect_delay, effect_attack, effect_fade,
+		(float)effect_volume / effect_fade);
+	*/
+
+	// FIFO for effects
+	last_haptic_effect_pos = (last_haptic_effect_pos + 1) % last_haptic_effect_size;
+	IN_Haptic_Effect_Shutdown(&last_haptic_effect[last_haptic_effect_pos].effect_id);
+	last_haptic_effect[last_haptic_effect_pos].effect_volume = effect_volume;
+	last_haptic_effect[last_haptic_effect_pos].effect_duration = effect_duration;
+	last_haptic_effect[last_haptic_effect_pos].effect_delay = effect_delay;
+	last_haptic_effect[last_haptic_effect_pos].effect_attack = effect_attack;
+	last_haptic_effect[last_haptic_effect_pos].effect_fade = effect_fade;
+	last_haptic_effect[last_haptic_effect_pos].effect_x = effect_x;
+	last_haptic_effect[last_haptic_effect_pos].effect_y = effect_y;
+	last_haptic_effect[last_haptic_effect_pos].effect_z = effect_z;
+	last_haptic_effect[last_haptic_effect_pos].effect_id = IN_Haptic_Effect_Init(
+		effect_x, effect_y, effect_z,
+		effect_duration, haptic_volume,
+		effect_delay, effect_attack, effect_fade);
+
+	return last_haptic_effect[last_haptic_effect_pos].effect_id;
+}
+
+// Keep it same with rumble rules
+static char *default_haptic_filter = (
+	// skipped files should be before wider rule
+	"!weapons/*grenlb "     // bouncing grenades don't have feedback
+	"!weapons/*hgrenb "     // bouncing grenades don't have feedback
+	"!weapons/*open "       // bouncing grenades don't have feedback
+	"!weapons/*warn "       // rogue's items don't have feedback
+	"weapons/ "             // any weapons that are not in previous list
+	// covered by previous rule
+	// "weapons/*plasshot " // phalanx cannon fire
+	// "weapons/*x "        // explosions...
+	// "weapons/*r "        // reloads & ion ripper fire
+	// "weapons/*blastf "   // dampen blasters
+	// "weapons/*hyprbf "   // dampen blasters
+	// "weapons/*nail "     // nailgun's fire
+	// "weapons/*shotgf "   // shotgun
+	// "weapons/*rocklf "   // RL
+	// "weapons/*sshotf "   // for super shotgun
+	// "weapons/*machgf "   // machine gun & disruptor fire
+	// "weapons/*disint "   // machine gun & disruptor fire
```

Comment:
I don't believe these comments serve a purpose here.
At line 1547, `"!weapons/*open "` should have the same comment as the line below: "rogue's items don't have feedback", since the `*open` sounds also come from rogue.